scylla/src/transport/topology.rs

```diff
@@ -189,16 +189,20 @@ impl MetadataReader {
 
         // shuffle known_peers to iterate through them in random order later
         self.known_peers.shuffle(&mut thread_rng());
-
-        let address_of_failed_control_connection = self.control_connection_address;
-        let filtered_known_peers = self
-            .known_peers
-            .iter()
-            .filter(|&peer| peer != &address_of_failed_control_connection);
+        debug!(
+            known_peers = self
+                .known_peers
+                .iter()
+                .fold(
+                    String::with_capacity(self.known_peers.len() * 20),
```

Comment:
in case somebody asks: 20 because I use the following peer string as reference: "127.127.127.127:9042". A more pessimistic one would also reserve 5 chars for the port instead of 6, but on the other hand the address strings are often shorter, and 20 is more round that 21, so it won.